File: onyx-database/src/com/onyx/diskmap/node/CombinedIndexHashMatrixNode.java
package com.onyx.diskmap.node;

/**
 * Created by tosborn1 on 1/10/17.
 *
 * This class indicates both a bitmap and skip list combined node
 *
 * @since 1.2.0
 */
public class CombinedIndexHashMatrixNode {

    public HashMatrixNode bitMapNode;
    public int hashDigit;
    public SkipListHeadNode head;

    public CombinedIndexHashMatrixNode(final SkipListHeadNode base, final HashMatrixNode node, int hashDigit) {
        this.head = base;
        this.bitMapNode = node;
        this.hashDigit = hashDigit;
    }

    @Override
    public int hashCode()
    {
        return Long.hashCode(this.bitMapNode.position);
    }

    @Override
    public boolean equals(Object object)
    {
        return (object instanceof CombinedIndexHashMatrixNode && ((CombinedIndexHashMatrixNode) object).bitMapNode.position == bitMapNode.position);
    }

}

File: onyx-database/src/main/kotlin/com/onyx/diskmap/data/CombinedIndexHashMatrixNode.kt
package com.onyx.diskmap.data

/**
 * Created by tosborn1 on 1/10/17.
 *
 * This class indicates both a bitmap and skip list combined data
 *
 * @since 1.2.0
 */
class CombinedIndexHashMatrixNode(var head: SkipListHeadNode, val bitMapNode: HashMatrixNode, val hashDigit: Int) {

    override fun hashCode(): Int = this.bitMapNode.position.hashCode()

    override fun equals(other: Any?): Boolean = other is CombinedIndexHashMatrixNode && other.bitMapNode.position == bitMapNode.position

}
